imap_processing/lo/l1c/lo_l1c.py

```diff
@@ -241,6 +246,52 @@ def create_pset_counts(
     return counts
 
 
+def calculate_exposure_times(counts: xr.DataArray, l1b_de: xr.Dataset) -> xr.DataArray:
+    """
+    Calculate the exposure times for the L1B Direct Event dataset.
+
+    The exposure times are calculated by binning the data into 3600 longitude bins,
+    40 latitude bins, and 7 energy bins. If more than one exposure time is in a bin,
+    the average is taken.
+
+    Parameters
+    ----------
+    counts : xarray.DataArray
+        An event counts array with dimensions (epoch, lon_bins, lat_bins, energy_bins).
+    l1b_de : xarray.Dataset
+        L1B Direct Event dataset. This data contains the average spin durations.
+
+    Returns
+    -------
+    exposure_time : xarray.DataArray
+        The exposure times for the L1B Direct Event dataset.
+    """
+    # Create bin edges
+    lon_edges = np.arange(3601)
+    lat_edges = np.arange(41)
+    energy_edges = np.arange(8)
+
+    data = np.column_stack(
+        (l1b_de["pointing_bin_lon"], l1b_de["pointing_bin_lat"], l1b_de["esa_step"])
+    )
+
+    result = binned_statistic_dd(
+        data,
+        l1b_de["avg_spin_durations"].to_numpy(),
```

Comment:
Wouldn't average spin duration be covering the entire 3600 points, so do you need to divide this up to get the exposure time per-pixel based upon the avg_spin_durations? Something like this over 3600 instead.